examples/joystick.py

```diff
@@ -76,24 +85,39 @@ def main():
         # Drawing step
         # First, clear the screen to white. Don't put other drawing commands
         # above this, or they will be erased with this command.
-        screen.fill("white")
+        screen.fill(WHITE)
         text_print.reset()
 
-        text_print.tprint(screen, f"Number of joysticks: {len(joysticks)}")
+        # Get count of joysticks.
+        joystick_count = pygame.joystick.get_count()
+
+        text_print.tprint(screen, f"Number of joysticks: {joystick_count}")
         text_print.indent()
 
         # For each joystick:
         for joystick in joysticks.values():
             jid = joystick.get_instance_id()
+            joystick.init()
+
+            try:
+                jid = joystick.get_instance_id()
+            except AttributeError:
+                # get_instance_id() is an SDL2 method
+                jid = joystick.get_id()
             text_print.tprint(screen, f"Joystick {jid}")
             text_print.indent()
 
-            # Get the name from the OS for the joystick.
+            # Get the name from the OS for the controller/joystick.
             name = joystick.get_name()
             text_print.tprint(screen, f"Joystick name: {name}")
 
-            guid = joystick.get_guid()
-            text_print.tprint(screen, f"GUID: {guid}")
+            try:
+                guid = joystick.get_guid()
+            except AttributeError:
+                # get_guid() is an SDL2 method
+                pass
```

Comment:
Same as above. We don't support SDL1 anymore, we write the examples for the current version. It should be fine without a try except?